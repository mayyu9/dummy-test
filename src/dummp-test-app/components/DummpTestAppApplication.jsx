import React from 'react';
import PropTypes from 'prop-types';
import { createStore, combineReducers, applyMiddleware, compose } from 'redux';
import { Provider } from 'react-redux';
import createSagaMiddleware from 'redux-saga';

import AppDelegate from 'terra-app-delegate';
import Application from 'terra-clinical-application';
import ContentContainer from 'terra-content-container';
import Header from 'terra-clinical-header';

import DummpTestAppContainer, { reducers, sagas } from './DummpTestAppContainer';

const sagaMiddleware = createSagaMiddleware();

const store = createStore(
  combineReducers(reducers),
  compose(applyMiddleware(sagaMiddleware)),
);

sagas.map(saga => (sagaMiddleware.run(saga)));

const propTypes = {
  app: AppDelegate.propType,
  locale: PropTypes.string.isRequired,
  patientId: PropTypes.number.isRequired,
  translationsLoadingHeader: PropTypes.string.isRequired,
  service: PropTypes.object,
};
k
// The disabling here is due to React-on-Rails not allowing default props
// Anyone mimicking this top level component should make it a class so that
// they can handle default props in the future.
// eslint-disable-next-line react/prefer-stateless-function
class DummpTestAppApplication extends React.Component {
  render() {
    const { app, locale, patientId, translationsLoadingHeader, service } = this.props;
    const translationsLoadingPlaceholder = (
      <ContentContainer header={<Header title={translationsLoadingHeader} />} />
    );

    return (
      <Provider store={store}>
        <Application locale={locale} app={app} translationsLoadingPlaceholder={translationsLoadingPlaceholder}>
          <DummpTestAppContainer
            patientId={patientId}
            service={service}
            key={'DUMMP_TEST_APP_APP'}
          />
        </Application>
      </Provider>
    );
  }
}

DummpTestAppApplication.propTypes = propTypes;

export default DummpTestAppApplication;
