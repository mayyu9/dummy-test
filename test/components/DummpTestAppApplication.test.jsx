import React from 'react';
import DummpTestAppApplication from '../../src/dummp-test-app/components/DummpTestAppApplication';

it('should render a DummpTestAppApplication with a placeholder', () => {
  const testDummyService = {
    getDummpTestApp: () => (
      new Promise((resolve) => {
        setTimeout(() => { resolve(); }, 1);
      })
    ),
  };

  const testView = shallow(<DummpTestAppApplication
    locale="en-US"
    patientId={100}
    service={testDummyService}
    translationsLoadingHeader="DummtTestEngine!!!!"
  />);
  expect(testView.first().shallow()).toMatchSnapshot();
});
