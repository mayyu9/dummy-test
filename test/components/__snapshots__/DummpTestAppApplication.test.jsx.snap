exports[`test should render a DummpTestAppApplication with a placeholder 1`] = `
<Application
  locale="en-US"
  translationsLoadingPlaceholder={
    <ContentContainer
      fill={false}
      header={
        <Header
          endContent={null}
          isSubheader={false}
          startContent={null}
          title="DummtTestEngine!!!!" />
      } />
  }>
  <Connect(DummpTestAppContainer)
    patientId={100}
    service={
      Object {
        "getDummpTestApp": [Function],
      }
    } />
</Application>
`;
